extensions/PhabBugz/lib/WebService.pm

```diff
@@ -66,10 +78,32 @@ sub check_user_permission_for_bug {
     };
 }
 
+sub check_user_enter_bug_permission {
+    my ($self, $params) = @_;
+
+    my $user = Bugzilla->login(LOGIN_REQUIRED);
+
+    $self->_validate_phab_user($user);
+
+    # Validate that a product name and user id are provided
+    ThrowUserError('phabricator_invalid_request_params')
+        unless ($params->{product} && $params->{user_id});
+
+    # Validate that the user exists
+    my $target_user = Bugzilla::User->check({ id => $params->{user_id}, cache => 1 });
+
+    # Send back an object with the attribute "result" set to 1 if the user
+    # can enter bugs into the given product, or 0 if not.
+    return {
+        result => $target_user->can_enter_product($params->{product}) ? 1 : 0
```

Comment:
Note: this will return 1 or 0, not true or false. That's probably fine, but if you wanted proper json booleans, you'd need to use `$self->type( 'boolean', $target_user->can_enter_product($params->{product}) )`

Reply:
Yeah I'm just following what check_user_permission_for_bug() does. I was wondering how to cast to a type though. :)

Reply:
The normal way is JSON->true and JSON->false, but `$self->type()` is what the rest of bugzilla uses.